plugins/woocommerce/includes/rest-api/Controllers/Version2/class-wc-rest-orders-v2-controller.php

```diff
@@ -289,6 +289,25 @@ function( $meta ) use ( $product, $order_item_name ) {
 			);
 		}
 
+		// Add additional applied coupon information.
+		if ( $item instanceof WC_Order_Item_Coupon ) {
+			$temp_coupon         = new WC_Coupon();
```

Comment:
Do we need to instantiate a coupon object here? I think we can just set the `data` keys using the `coupon_info ` or  `coupon_data` directly.

Reply:
Using an actual coupon object allows the info encoding and deconding logic to be constrained to a single place (the coupon class), which is good for code readability and robustness.